src/qibo_cloud_backends/qiskit_client.py

```diff
@@ -16,18 +17,24 @@ class QiskitClientBackend(NumpyBackend):
         platform (str): The IBM platform.
     """
 
-    def __init__(self, token, provider="ibm-q", platform="ibmq_qasm_simulator"):
+    def __init__(self, token, provider=None, platform=None):
         super().__init__()
+        if not provider:
+            provider = "ibm-q"
+        if not platform:
+            platform = "ibmq_qasm_simulator"
```

Comment:
Actually why? It is usually accepted to have strings as defaults (despite being mutable).

However, if you want to keep it like this, you should write `if provider is not None`, to avoid relying on the falsiness of `None` (even `0` is a falsy value, but it's definitely different from `None`).

Reply:
Because of the `set_backend` method of qibo, that one has `None` as default values for each argument. That `None` therefore overwrites any default value give here to provider and platform I believe.

Reply:
I see... then just use the `is not`. And the same applies for the TII provider.